src/dependency_definition.cr

```diff
@@ -45,59 +45,51 @@ module Shards
     #
     # Split to allow better unit testing.
     def self.parts_from_cli(value : String) : Parts
-      resolver_key = nil
-      source = ""
-      requirement = Any
-
-      if value.starts_with?("file://")
-        resolver_key = "path"
-        source = value[7..-1] # drop "file://"
-      end
-
-      # relative paths
-      path = Path[value].to_posix.to_s
-      if path.starts_with?("./") || path.starts_with?("../")
-        resolver_key = "path"
-        source = path
-      end
-
       uri = URI.parse(value)
-      if uri.scheme != "file" && uri.host &&
-         (resolver_key = GitResolver::KNOWN_PROVIDERS[uri.host]?)
-        source = uri.path[1..-1].rchop(".git") # drop first "/""
-      end
-
-      if value.starts_with?("git://")
-        resolver_key = "git"
-        source = value
-      end
-
-      if value.starts_with?("git@")
-        resolver_key = "git"
-        source = value
-      end
-
-      unless resolver_key
-        Resolver.resolver_keys.each do |key|
-          key_schema = "#{key}:"
-          if value.starts_with?(key_schema)
-            resolver_key = key
-            source = value.sub(key_schema, "")
-
-            # narrow down requirement
-            if source.includes?("@")
-              source, version = source.split("@")
-              requirement = VersionReq.new("~> #{version}")
-            end
 
-            break
+      case scheme = uri.scheme
+      when Nil
+        case value
+        when .starts_with?("./"), .starts_with?("../")
+          Parts.new("path", Path[value].to_posix.to_s)
+        when .starts_with?("git@")
```

Comment:
Edge case: it's the username to connect with, it doesn't have to be `git`.

Reply:
Yes of course. This is just an example for a heuristic. If the username is `git`, it's likely a git URL.

We can do similar things for other URI components. For example, if the hostname starts with `fossil.`, it's likely a fossil URL. If the path ends with `.git` it's likely a git URL.

I'm not entirely sure if we should do all those things when it's relatively simple to just provide a fully qualified URI. But it seems easy enough, shouldn't cause much issues and might improve UX in some cases.